Dockerfile

```
@@ -4,14 +4,16 @@ FROM ubuntu:22.04
 RUN apt-get update \
     && apt-get upgrade -y \
     && apt-get install -y build-essential curl libffi-dev libffi8ubuntu1 libgmp-dev libgmp10 libncurses-dev libncurses5 libtinfo5 libnuma1 git libnuma-dev llvm zlib1g-dev \
-    && curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+RUN useradd -m haskell
+USER haskell
+RUN curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | \
     BOOTSTRAP_HASKELL_NONINTERACTIVE=1 \
     BOOTSTRAP_HASKELL_ADJUST_BASHRC=1 \
     BOOTSTRAP_HASKELL_MINIMAL=1 \
-    sh \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/*
-
-ENV PATH="${PATH}:/root/.ghcup/bin"
+    sh
 
-RUN ghcup install stack
\ No newline at end of file
+ENV PATH=$PATH:/home/haskell/.ghcup/bin
+RUN /home/haskell/.ghcup/bin/ghcup install stack
+WORKDIR /haskell
```

because we've managed to set PATH properly, we don't need to call the full path

and `/work` to match `docker run` command the README

```suggestion
ENV PATH=$PATH:/home/haskell/.ghcup/bin
RUN ghcup install stack
WORKDIR /work
```